src/java.base/share/classes/java/io/Console.java

```diff
@@ -145,6 +146,124 @@ public Reader reader() {
         throw newUnsupportedOperationException();
     }
 
+    /**
+     * Writes a string representation of the specified object to this console's
+     * output stream, terminates the line and then flushes the console.
+     *
+     * <p> The effects of {@code con.println(obj)} are as if by calling
+     * {@snippet lang = java:
+     *     con.format("%s%n", obj)
+     * }
```

Comment:
I don't think we need to specify this in terms of `format("%s")`. It should be sufficient to say that the string representation is written to the console's output stream.

Reply:
Okay. Maybe we don't need to tie new `print(Object)`/`println(Object)` to `format(String, Object...)` like old `printf(String, Object...)` does. But it would be good to remove uncertainty on their behaviour by maybe reducing those new methods to some pre-existing ones.

For example, it's not obvious that a "string representation" of `null` is `"null"` and not NPE; or that a "string representation" of `new char[]{'a'}` is something like `[C@133314b` rather than `"a"`, which it is for `System.out.println(char[])`.

Reply:
Right, when you and I discussed this I think we gravitated toward following `String.valueOf(Object)`.